Remove the consumption and production power formulas

matthias-wende-frequenz opened the pull request:
Users must understand passive sign conventions to utilize these formulas effectively, and therefore they do not provide significant benefits.

matthias-wende-frequenz committed:
Remove the cons/prod formulas from logical meter

These formulas are replaced by a combination of the corresponding
passive sign convention formulas and the recently added consumption
and production operators.

Signed-off-by: Matthias Wende <matthias.wende@frequenz.com>

cwasicki commented:
> Users must understand passive sign conventions to utilize these formulas effectively, and therefore they do not provide significant benefits.

What do you mean? These formulas were introduced exactly for the reason that passive sign convention can be confusing.

matthias-wende-frequenz committed:
Update release notes

Signed-off-by: Matthias Wende <matthias.wende@frequenz.com>

shsms commented:
I will leave it for @cwasicki to approve, because he has open questions, but if we are doing this, we should also remove the corresponding formulas from the battery pool and the ev charger pool.

matthias-wende-frequenz commented:
# Possible approaches for keeping the functionality alive

We discussed that these formulas are useful but they might lead to subtle bugs when confusing the slightly different functions.
We can provide this by adding a production and consumption operators as suggest in the following.

## 1

chp_power("production")
chp_power("abs_production")
chp_power().production() or production_op(chp_power())

## 2

chp_power().production() and chp_power().consumption()

llucax commented:
Option 2 wins! BTW, Option 1.3 is Option 2, right?

cwasicki commented:
>Option 1.3 is Option 2, right?

Yes, but option 1.3a only ;-)

matthias-wende-frequenz commented:
Back to draft as it needs the operators to get implemented.

matthias-wende-frequenz committed:
Remove prod/cons formulas from battery pool

Signed-off-by: Matthias Wende <matthias.wende@frequenz.com>

matthias-wende-frequenz committed:
Remove FormulaType

The formula Type has been used to distinguish between
consumption, production or passive sign convention formulas.
With the recent changes the sdk only provides passive sign
convention formulas.

Signed-off-by: Matthias Wende <matthias.wende@frequenz.com>

matthias-wende-frequenz committed:
Remove prod/cons formulas from ev pool

Signed-off-by: Matthias Wende <matthias.wende@frequenz.com>

matthias-wende-frequenz commented:
unblocked after #746 has been merged as a replacement

llucax commented on tests/timeseries/test_logical_meter.py:
```diff
@@ -427,14 +377,14 @@ async def test_consumer_power_2_grid_meters(
 
         logical_meter = microgrid.logical_meter()
-        grid_consumption_recv = logical_meter.grid_consumption_power.new_receiver()
+        grid_consumption_recv = logical_meter.grid_power.new_receiver()
```
I guess you are purposely not using `consumption` (with the new operator) anymore here because this is already tested in the new operator tests, right?

llucax reviewed (dismissed):
Yeeeiii! Removing code and special cases! :tada:

Marenz commented:
Some of the [commits](https://github.com/frequenz-floss/frequenz-sdk-python/pull/697/commits/5b25cf3bf2541d84bdae37f66ad0ae1f43f7e8ea) here are a great example of unhelpful commit title/messages. 
Just reading the commit I had no idea why it was doing what it was doing "Remove xyz"..great.. but why?!
I had to find this PR first to get an idea of why it was happening.

Please, write the why, not the what. It was very very obvious that it was removing something... no need to mention that even imo.

matthias-wende-frequenz commented:
> Just reading the commit I had no idea why it was doing what it was doing "Remove xyz"..great.. but why?!

The reason is written up in the first of the commit chain.

> These formulas are replaced by a combination of the corresponding
passive sign convention formulas and the recently added consumption
and production operators.

Next time I'll add a note to each of the involved commits ;).

Marenz commented:
> The reason is written up in the first of the commit chain.

right, thus I said "some" ;) but I still had to do more investigation to be sure. 

> Next time I'll add a note to each of the involved commits ;).

<3
